packages/aggregator/src/analytics/aggregateAnalytics/aggregations/getFinalValuePerPeriod.js

```diff
@@ -106,12 +106,19 @@ class FinalValueAggregator {
     this.cache = cache;
   }
 
-  getContinuousValues(analytics, aggregationPeriod) {
-    const periods = getContinuousPeriodsForAnalytics(analytics, aggregationPeriod, true);
+  getFilledValues(analytics, aggregationPeriod, fillEmptyPeriodsWith) {
+    const checkIfFillingContinuousValues = fillEmptyPeriodsWith === 'previous';
+    const periods = getContinuousPeriodsForAnalytics(
+      analytics,
+      aggregationPeriod,
+      checkIfFillingContinuousValues, // To get a continuous period till now, we need to set true for this option config (continueTilCurrentPeriod). This can be a new option config in the future.
+    );
 
     const values = [];
     this.cache.iterateOrganisationUnitCache(organisationUnitCache => {
```

Comment:
```suggestion
    const values = [];
    this.cache.iterateOrganisationUnitCache(organisationUnitCache => {
      const periods = getContinuousPeriodsForAnalytics(
        Object.values(organisationUnitCache),
        aggregationPeriod,
        checkIfFillingContinuousValues, // To get a continuous period till now, we need to set true for this option config (continueTilCurrentPeriod). This can be a new option config in the future.
```

Would something like this work to address your comment above? https://github.com/beyondessential/tupaia/pull/2471/files#r627825280

Reply:
Do you mean test cases can explain the logic and we can remove this comment?